### PR TITLE
ARROW-13976: [C++] Add path to libjvm.so in ARM CPU

### PR DESCRIPTION
**Pull request by wuzhuoming**
resolve issue ARROW-13976, add path in hdfs_internal.cc to find libjvm.so in ARM CPU.

### CODE REVIEW

**Comment by pitrou** [comment]
@cyb70289 Would you know if this looks good?

File: cpp/src/arrow/io/hdfs_internal.cc [hunk 1]
```diff
@@ -215,7 +215,7 @@ Result<std::vector<PlatformFilename>> get_potential_libjvm_paths() {
       }));
   ARROW_ASSIGN_OR_RAISE(search_suffixes,
                         MakeFilenameVector({"", "/jre/lib/amd64/server",
-                                            "/lib/amd64/server", "/lib/server"}));
+                                            "/lib/amd64/server", "/lib/server","/jre/lib/aarch64/server"}));
```
[the remark this answers is not in the document]
Comment (cyb70289) [reply cpp/src/arrow/io/hdfs_internal.cc:218]: Mixing `amd64` and `aarch64` is one list looks not good.
We can define the `arch` string per compiler architecture and do `"/jre/lib/" + arch + "/server"`.

**Review by cyb70289 — CHANGES_REQUESTED** [review]

File: cpp/src/arrow/io/hdfs_internal.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (wuzhuoming) [reply cpp/src/arrow/io/hdfs_internal.cc]: I have updated the prefix.
But for the suffixes, if defining a new string arch to  select the  compiler architecture, could you please tell me how to do it? I am not very familiar with CPP.

[the remark this answers is not in the document]
Comment (cyb70289) [reply cpp/src/arrow/io/hdfs_internal.cc]: Something like this:

```
#if defined(__aarch64__)
const std::string prefix_arch{"arm64"};
const std::string suffix_arch{"aarch64"};
#else
const std::string prefix_arch{"amd64"};
const std::string suffix_arch{"amd64"};
#endif
ARROW_ASSIGN_OR_RAISE(
      search_prefixes,
      MakeFilenameVector({
          ......
          "/usr/local/lib/jvm/java-8-openjdk-" + prefix_arch,  // alt ubuntu / debian distros
          "/usr/lib/jvm/java-8-openjdk-" + prefix_arch,        // alt ubuntu / debian distros
          ......
      }));
  ARROW_ASSIGN_OR_RAISE(search_suffixes,
                        MakeFilenameVector({"", "/lib/server",
                                            "/jre/lib/" + suffix_arch + "/server", "/lib/" + suffix_arch + "/server"}));
  file_name = "libjvm.so";
```

[the remark this answers is not in the document]
Comment (wuzhuoming) [reply cpp/src/arrow/io/hdfs_internal.cc]: Thanks a lot for your instruction! I had done the modification of code, please check.

File: cpp/src/arrow/io/hdfs_internal.cc [hunk 2]
```diff
@@ -212,10 +219,16 @@ Result<std::vector<PlatformFilename>> get_potential_libjvm_paths() {
           "/usr/local/lib/jvm/java-6-oracle",         // alt ubuntu
           "/usr/lib/jvm/default",                     // alt centos
           "/usr/java/latest",                         // alt centos
+          "/usr/local/lib/jvm/java-1.8.0-openjdk-"+prefix_arch,  // debian distros / arm64 architecture
+          "/usr/lib/jvm/java-1.8.0-openjdk-"+prefix_arch,        // debian distros / arm64 architecture
+          "/usr/local/lib/jvm/java-1.7.0-openjdk-"+prefix_arch,  // debian distros / arm64 architecture
+          "/usr/lib/jvm/java-1.7.0-openjdk-"+prefix_arch,        // debian distros / arm64 architecture
+          "/usr/local/lib/jvm/java-1.6.0-openjdk-"+prefix_arch,  // debian distros / arm64 architecture
+          "/usr/lib/jvm/java-1.6.0-openjdk-"+prefix_arch        // debian distros / arm64 architecture
```
Comment (cyb70289) [inline cpp/src/arrow/io/hdfs_internal.cc:227]: I don't think these added lines are necessary.

Comment (wuzhuoming) [reply cpp/src/arrow/io/hdfs_internal.cc]: Maybe for those who use openjdk?  Anyway, I had removed these line, thtank you for your reminding.

**Comment by cyb70289** [comment]
MinGW failures tracked at https://issues.apache.org/jira/browse/ARROW-13999

**Review by cyb70289 — APPROVED** [review]
+1

**Comment by cyb70289** [comment]
@pitrou, do you have comments?

**Comment by pitrou** [comment]
No :-)

**Comment by cyb70289** [comment]
Thanks @wuzhuoming !

**Pull request closed without merging**